option.go

```diff
@@ -251,18 +254,19 @@ type compressionOption struct {
 }
 
 func (o *compressionOption) applyToClient(config *clientConfig) {
-	o.apply(config.CompressionPools)
+	if o.Name == "" || o.CompressionPool == nil {
+		return
+	}
+	config.CompressionPools[o.Name] = o.CompressionPool
+	config.CompressionNames = append(config.CompressionNames, o.Name)
 }
 
 func (o *compressionOption) applyToHandler(config *handlerConfig) {
-	o.apply(config.CompressionPools)
-}
-
-func (o *compressionOption) apply(m map[string]*compressionPool) {
 	if o.Name == "" || o.CompressionPool == nil {
 		return
 	}
-	m[o.Name] = o.CompressionPool
+	config.CompressionPools[o.Name] = o.CompressionPool
```

Comment:
Just jumped out at me as a potential NPE, but I do know we always make sure this is initialized right?

Reply:
Yep! Always initialized.